src/base/multithreading.cc

```diff
@@ -65,6 +84,19 @@ bool is_logic_thread() {
 	return logic_thread == std::this_thread::get_id();
 }
 
+static std::string thread_name(const std::thread::id id) {
+	if (id == kNoThread) {
+		return "No thread";
+	}
+	if (id == initializer_thread) {
+		return "Initializer thread";
+	}
+	if (id == logic_thread) {
+		return "Logic thread";
+	}
+	return "Auxiliary thread";  // We don't have that many threads currently...
```

Comment:
we still could use a switch :-)

Reply:
Wouldn't work because `logic_thread` and `initializer_thread` are variables set at runtime. `switch-case` only works with `constexpr`s that can be evaluated at compile time.